### PR TITLE
feat: app.close return a promise

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

app.close

### CODE REVIEW

**Comment by atian25** [comment]
+1

File: lib/egg.js [hunk 1]
```diff
@@ -119,11 +119,19 @@ class EggCore extends KoaApplication {
...
    * close all listeners
    * @member {Function}
+   * @return {Promise} promise
    * @since 1.0.0
    */
   close() {
-    this.emit('close');
-    this.removeAllListeners();
+    return new Promise((resolve, reject) => {
+      try {
```
Comment (dead-horse) [inline lib/egg.js]: 意义在哪里呢？我看现在这是一个同步的方法啊？

Comment (dead-horse) [inline lib/egg.js]: 为什么要return promise

Comment (dead-horse) [inline lib/egg.js]: 现在也可以这样写啊，close 是同步的

Comment (popomore) [inline lib/egg.js]:
```js
close() {
  return super
    .close()
    .then(resolve => {
      // close sth
      resolve();
    });
}
```

**Comment by popomore** [comment]
Closes egg/egg#115

**Comment by fengmk2** [comment]
+1

**Pull request merged**